packages/react-magma-dom/src/components/CharacterCounter/CharacterCounter.test.js

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+import { axe } from '../../../axe-helper';
+import { CharacterCounter } from '.';
+import { render, fireEvent } from '@testing-library/react';
+import { I18nContext } from '../../i18n';
+import { defaultI18n } from '../../i18n/default';
+
+const testId = 'test-id';
+
+const characterAllowed = defaultI18n.characterCounter.characterAllowed;
+
+const charactersAllowed = defaultI18n.characterCounter.charactersAllowed;
+
+const characterLeft = defaultI18n.characterCounter.characterLeft;
+
+const charactersLeft = defaultI18n.characterCounter.charactersLeft;
+
+const characterOver = defaultI18n.characterCounter.characterOver;
+
+const charactersOver = defaultI18n.characterCounter.charactersOver;
+
+describe('CharacterCounter', () => {
+  it('should find element by testId', () => {
+    const { getByTestId } = render(
+      <CharacterCounter maxLength={231} testId={testId} />
+    );
+
+    expect(getByTestId(testId)).toBeInTheDocument();
+  });
+
+  it('Does not violate accessibility standards', () => {
+    const { container } = render(<CharacterCounter maxLength={22} />);
+
+    return axe(container.innerHTML).then(result => {
+      return expect(result).toHaveNoViolations();
+    });
+  });
+  describe('Titles', () => {
+    const labelText = 'Character Counter';
+
+    describe('Characters Allowed', () => {
+      it('Shows the default label of "characters allowed" if maxLength is 0', () => {
+        const { getByText } = render(
+          <I18nContext.Provider
+            value={{
+              ...defaultI18n,
+            }}
+          >
+            <CharacterCounter maxLength={0} />
+          </I18nContext.Provider>
+        );
+
+        expect(getByText('0 ' + charactersAllowed)).toBeInTheDocument();
+      });
+
+      it('Shows the default label of "character allowed" if maxLength is 1', () => {
+        const { getByText } = render(
+          <I18nContext.Provider
+            value={{
+              ...defaultI18n,
+            }}
+          >
+            <CharacterCounter maxLength={1} />
+          </I18nContext.Provider>
+        );
+
+        expect(getByText('1 ' + characterAllowed)).toBeInTheDocument();
+      });
+      it('Shows the default label of "characters allowed"', () => {
+        const { getByText } = render(
+          <I18nContext.Provider
+            value={{
+              ...defaultI18n,
+            }}
+          >
+            <CharacterCounter maxLength={2} />
+          </I18nContext.Provider>
+        );
+
+        expect(getByText('2 ' + charactersAllowed)).toBeInTheDocument();
+      });
+    });
+
+    describe('Characters Left', () => {
+      it('Shows the label "characters left" as the user types', () => {
```

Comment:
So here we aren't testing the user typing. This component has a prop `inputLength` that just gives it the length of the input. So a better description would be something like `Shows the label "characters left" when inputLength > 1`